### PR TITLE
feat(developer): kmc-kmn

### PR DESCRIPTION
**Pull request by mcdurdin**
Compile a .kmn keyboard to .kmx, Typescript module.

Builds on kmcmplib WASM build and accessible through command line with kmc.

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc-kmn/build.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# adjust relative paths as necessary
+THIS_SCRIPT="$(readlink -f "${BASH_SOURCE[0]}")"
+. "${THIS_SCRIPT%/*}/../../../resources/build/build-utils.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
+
+cd "$THIS_SCRIPT_PATH"
+
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+
+builder_describe "Build Keyman kmc kmn to kmx Compiler module" \
```
Comment (jahorton) [inline developer/src/kmc-kmn/build.sh:19]: Three three-letter file-extension-looking names in four tokens (all starting with `km`, no less!) makes for a pretty "fun" parse.  Nothing critical here, but I think this (or something similar) might be a tad more readable...

```suggestion
builder_describe "Build Keyman's kmc tool's kmn-to-kmx Compiler module" \
```

Comment (jahorton) [reply developer/src/kmc-kmn/build.sh:19]: Alternatively...

```suggestion
builder_describe "Builds Keyman Developer's kmn keyboard compiler module" \
```

would be a far better match for the package.json's name field.

File: developer/src/kmc-kmn/src/compiler/compiler.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
+//
+// const importObject = {
+  // imports: { imported_func: (arg) => console.log(arg) },
+// };
```
Comment (jahorton) [inline developer/src/kmc-kmn/src/compiler/compiler.ts:4]: Someone usually tells me to eliminate code commented like this... is there a specific reason we're keeping it around like this?

File: developer/src/kmc-kmn/test/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  "compilerOptions": {
+      "rootDir": ".",
+      "rootDirs": ["./", "../src/"],
+      "outDir": "../build/test",
+      "baseUrl": ".",
+      "allowSyntheticDefaultImports": true, // for chai
+      "paths": {
+        // "@keymanapp/keyman-version": ["../../../common/web/keyman-version/keyman-version.mts"],
+        "@keymanapp/common-types": ["../../../../common/web/types/src/main"],
+        // "@keymanapp/": ["core/include/ldml/ldml-keyboard-constants"],
```
Comment (jahorton) [inline developer/src/kmc-kmn/test/tsconfig.json:13]: Got a couple of commented tsconfig.json entries here.

File: developer/src/kmc-kmn/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{
+  "extends": "../../../tsconfig.esm-base.json",
+
+  "compilerOptions": {
+    "outDir": "build/src/",
+    "rootDir": "src/",
+    "baseUrl": ".",
+    "allowJs": true,
+    "paths": {
+      // "@keymanapp/keyman-version": ["../../../common/web/keyman-version/keyman-version.mts"],
```
Comment (jahorton) [inline developer/src/kmc-kmn/tsconfig.json:10]: Another one here.

File: developer/src/kmc-kmn/build.sh [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+if builder_start_action test; then
+  npm test
+  builder_finish_action success test
+fi
+
+#-------------------------------------------------------------------------------------------------------------------
+
+if builder_start_action publish; then
+  . "$KEYMAN_ROOT/resources/build/npm-publish.inc.sh"
+  npm_publish
```
Comment (darcywong00) [inline developer/src/kmc-kmn/build.sh:72]: Did you want to check for builder_option `dry-run`?

File: developer/src/kmc-kmn/test/fixtures/000.kmn [hunk 6]
```diff
@@ -0,0 +1,11 @@
+c Description: Tests null keyboard
```
Comment (darcywong00) [inline developer/src/kmc-kmn/test/fixtures/000.kmn:1]: I noticed on #8529 you called this `k_000___null_keyboard.kmn`.

Is the plan to consolidate using common/test/keyboards/?

Comment (mcdurdin) [reply developer/src/kmc-kmn/test/fixtures/000.kmn:1]: Good catch, will do this in a follow-up PR

File: developer/src/kmc-kmn/build.sh (hunk 5 above)
Comment (mcdurdin) [reply developer/src/kmc-kmn/build.sh:72]: Good catch, will do this in a follow-up PR

File: core/tests/unit/kmx/meson.build [hunk 7]
```diff
@@ -89,29 +89,22 @@ tests = [
...
-    kmcomp_cmd = [wine, kmcomp]
-    if not wine.found()
-      kmcomp = disabler()
-    endif
-  endif
   copy_cmd = [find_program('cp', required: true)]
   cat_cmd = [find_program('cat', required: true)]
 endif
 
 if cpp_compiler.get_id() == 'emscripten'
```
Comment (darcywong00) [inline core/tests/unit/kmx/meson.build:105]: Is it worth assigning this to a boolean? I see check happens 5x in the file.

**Review by darcywong00 — APPROVED** [review]
lgtm

turning a blind eye to all the test builds :)

File: core/tests/unit/kmx/meson.build (hunk 7 above)
Comment (mcdurdin) [reply core/tests/unit/kmx/meson.build:105]: I think I will be doing some more cleanup on the meson.build files in coming weeks, so will keep this in consideration

**Pull request merged**